### ISSUE TITLE
[FEATURE REQUEST] Allow Vault module to fetch specific version of a secret

### ISSUE THREAD

**Issue by mdschmitt** [saltstack/salt#60758]
**Is your feature request related to a problem? Please describe.**
As it stands, the `vault.read_secret` module fetches whatever the latest version of a secret is from Hashicorp Vault.  The KV2 secrets engine allows for versioned secrets, which is useful if there are differing values that should be stored in the same path (such as sensitive config values per-application or environment).

**Describe the solution you'd like**
Ideally, there would be a way to specify the desired key _and_ the version of the secret so that different values can be fetched without needing to add new paths to Vault or update statefiles to look in different locations.

**Describe alternatives you've considered**
Custom modules such as those at https://backbeat.tech/blog/secure-servers-with-saltstack-and-vault-part-3

**Additional context**
N/A

**Comment by OrangeDog** [comment]
I don't think versioning is supposed to be used for per-app config. They should be a different paths.
It's just for auditing / rollback isn't it?

**Comment by mdschmitt** [comment]
You're right.  I guess the reason I thought this could be useful was because I tied my shoes together and didn't realize I wouldn't be able to undo it.

As it stands, if you write `/secrets/salt/mysecret` (version 1) to contain:
```
foo=bar
bar=baz
```

and then later write `/secrets/salt/mysecret` (version 2) as:
```
foo=baz
baz=foo
```

and then run your salt state, you now no longer have the `bar` value and there's no way to go get it from Salt's perspective (even though it's still there in Vault).  At that point, you have to go to Vault and `vault kv get` the "bar" value.  Problematic if you don't know Salt's approle creds and don't have a Vault access token with a policy allowed to read /secrets/salt/mysecret

Possible I'm just using Vault's secret engine incorrectly.  

Separate question: is there code anywhere that allows minions to generate and use one-time creds or tokens instead of just vanilla grabbing a k/v secret val?  Part of Vault's power is the protection of secrets via temp creds, ACLs and TTLs.  If the real db password just ends up in plaintext in the config file anyway, I get sort of a "what was the point?" feeling..
